README.md

```diff
@@ -6,18 +6,44 @@
 **Warning: This is a work in progress, it is not yet usable.**
 
 A standalone tool that converts [Clang-Tidy](https://clang.llvm.org/extra/clang-tidy/)
-diagnostics to Codacy's format. It allows the integration of Clang-Tidy into your Codacy workflow.
+diagnostics to Codacy's format.
 
-TODO CY-629: Add link explaining how custom tools integrate with codacy
+It allows running Clang-Tidy either locally or as part of your CI process and then integrating the results into your Codacy workflow. This way, Codacy will present the results coming from Clang-Tidy alongside all the other code quality information in the dashboards.
+
+Currently, you can only use codacy-clang-tidy on Codacy self-hosted installations.
 
 ## Usage
 
-The upload of results for a commit is done in two steps:
+### Requirements
+
+To get your Clang-Tidy results into Codacy you'll need to:
+
+-   Enable the setting “Run analysis through build server” under your repository Settings > General > Repository analysis
+-   Obtain a [Project API token](https://support.codacy.com/hc/en-us/articles/207994675-Project-API)
+-   Download [codacy-clang-tidy](https://github.com/codacy/codacy-clang-tidy/releases)
+
+### Sending the results to Codacy
+
+Sending the results of running Clang-Tidy to Codacy involves the steps below, which you can automate in your CI build process:
+
+1.  Run Clang-Tidy
+2.  Convert the Clang-Tidy output to a format that the Codacy API accepts
+3.  Send the results to Codacy
+4.  Finally, signal that Codacy can use the sent results and start a new analysis
+
+With script:
+
+```bash
+export PROJECT_TOKEN="YOUR-TOKEN"
+export COMMIT="COMMIT-UUID"
+export CODACY_URL="CODACY-INSTALLATION-URL" # if not defined https://api.codacy.com will be used
+export CODACY_CLANG_TIDY_VERSION=0.2.3 # if not defined, latest will be used
 
--   uploading all results
--   telling Codacy that it can run the rest of the analysis
+clang-tidy "<clang-tidy-configs>" | \
+./<codacy-clang-tidy-path>/scripts/send-results.sh # requires a codacy-clang-tidy-"<version>" in the current directory
```

Comment:
This is nice and simplifies things a lot!

Will we consider having this way of sending the results back to Codacy as the default? Meaning:

* We no longer have to explain the intermediate steps above nor have an example code block detailing all the commands
* codacy-clang-tidy-<version> would actually include the calls to the API so we wouldn't even need an extra script to act as a wrapper

Does this make sense?

Reply:
I would leave the other ones just so that users can easily understand what the script is doing behind the scenes.
Later, they can do the same for windows or something different.
What do you think? @mrfyda do you agree with this idea?

Reply:
it's looking great